app/code/Magento/SalesRule/Model/ResourceModel/Rule/Customer.php

```diff
@@ -48,4 +48,22 @@ public function loadByCustomerRule($rule, $customerId, $ruleId)
         $rule->setData($data);
         return $this;
     }
+
+    /**
+     * Delete the time Usage from salesrule_customer table when times_used is 0
+     * @param int $ruleId
+     * @param int $customerId
+     * @param int $updatedTimeUsed
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function deleteCustomerTimeUsage($ruleId, $customerId, $updatedTimeUsed)
```

Comment:
Consider moving this logic to separate class and inject it instead of the resource.

Reply:
can you suggest if any existing class can be used to move the logic into or should I create a new class?

Reply:
Hi @Renu05 , it would be the best to create new class for this logic

Reply:
Hi @sidolov, I have created a new class and move the delete logic in DeleteCustomerUsage file. kindly check and let me know if any further modification needed.

Reply:
hi @sidolov, any update on this.